src/node/desktop/src/main/gwt-window.ts

```diff
@@ -25,6 +25,16 @@ export abstract class GwtWindow extends DesktopBrowserWindow {
   constructor(options: WindowConstructorOptions) {
     super(options);
     this.window.on('focus', this.onActivated.bind(this));
+
+    // when a GWT window is navigated (for example, because we opened a new project),
+    // we need to clear the history -- otherwise, Electron may attempt to handle the
+    // mouse forward + backward buttons and "navigate" to the previously-open project,
+    // which will fail
+    //
+    // https://github.com/rstudio/rstudio/issues/11016
+    this.window.webContents.on('did-navigate', (event, params) => {
```

Comment:
I _think_ we only want this behavior in GWT windows (that is, the main window + "satellite" windows); however, we might want to allow forward + back navigation in other windows. Does that sound right?

Reply:
Yes, I think that's right.